lua/lint/linters/mlint.lua

```diff
@@ -0,0 +1,43 @@
+local pattern = "L (%d+) %(C (%d+)[-]?([%d]*)%): (.-): (.-): (.*)"
+local severities = {
+  ML2 = vim.lsp.protocol.DiagnosticSeverity.Error,
+  ML3 = vim.lsp.protocol.DiagnosticSeverity.Error,
+  ML4 = vim.lsp.protocol.DiagnosticSeverity.Error,
+  ML1 = vim.lsp.protocol.DiagnosticSeverity.Warning,
+  ML0 = vim.lsp.protocol.DiagnosticSeverity.Information,
+  ML5 = vim.lsp.protocol.DiagnosticSeverity.Hint,
+}
+
+return {
+  cmd = "mlint",
+  stdin = false,
+  stream = "stderr",
+  args = { "-cyc", "-id", "-severity" },
+  ignore_exitcode = true,
+  parser = function(output, _)
+    local result = vim.fn.split(output, "\n")
+    local diagnostics = {}
+
+    for _, line in ipairs(result) do
+      local lineno, start_col, end_col, code, sev, desc = string.match(line, pattern)
```

Comment:
Given that this uses `string.match` with a pattern, would it be possible to use the `from_pattern` function?

See `pycodestyle`, `mypy`, `eslint` or `clangtidy` for examples.

Reply:
I think it might be possible...

Reply:
Thanks! I ended up using the same logic as `from_pattern`.